src/providers/pkcs11_provider/mod.rs

```diff
@@ -493,18 +501,13 @@ impl Provide for Pkcs11Provider {
     ) -> Result<psa_generate_key::Result> {
         info!("Pkcs11 Provider - Create Key");
 
-        if op.attributes.key_type != KeyType::RsaKeyPair
-            || op.attributes.key_policy.key_algorithm
-                != Algorithm::AsymmetricSignature(AsymmetricSignature::RsaPkcs1v15Sign {
-                    hash_alg: Hash::Sha256,
-                })
-        {
-            error!(
-                "The PKCS11 provider currently only supports creating RSA key pairs for signing and verifying. The signature algorithm needs to be RSA PKCS#1 v1.5 and the text hashed with SHA-256.");
+        if op.attributes.key_type != KeyType::RsaKeyPair {
```

Comment:
This might annoy you, but could we move this check somewhere else, so that it's "global"? Maybe in the interface or in `providers/mod.rs`, where we can just have a list of supported algos that we can add to as we go. Or does that not work because we'd need to implement them everywhere at the same time

Reply:
I'm just thinking that we'll keep adding stuff, this `if` will keep growing

Reply:
I was thinking about putting those checks in common put it is not an easy problem:

* we probably want indepentant checks per provider as we might implement key attributes/algs support differently in each provider
* those checks can be different in each functions: the supported key attributes can be different in generate and import_key for example. For operations where an algorithm is specified, different operations might support different algorithms in the same provider...

I see your point though that those conditions everywhere might be messy.